shell_sort/ShellSort.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import "fmt"
+
+// Worst case time complexity = O(n^2)
+// Best case complexity = O(nlog(n))
+// shellSort returns sorted data
+func shellSort(data []int) []int {
```

Comment:
It would be better to have this function exported i.e. start with a capital. (ShellSort) 
This will allow people to use this file as a module and call ShellSort from their code.